docroot/modules/custom/mass_content/src/MassContentBatchManager.php

```diff
@@ -142,6 +142,70 @@ public static function processServiceNode($id, ContentEntityBase $node, array $s
     );
   }
 
+  /**
+   * Process contextual search related field default values.
+   */
+  public static function processContextualSearchFields($id, ContentEntityBase $node, bool $do_not_include, bool $show_parent, $operation_details, &$context) {
+    // Don't spam all the users with content update emails.
+    $_ENV['MASS_FLAGGING_BYPASS'] = TRUE;
+
+    // Turn off entity_hierarchy writes while processing the item.
+    \Drupal::state()->set('entity_hierarchy_disable_writes', TRUE);
+
+    try {
+      if ($do_not_include) {
+        $node->set('field_org_no_search_filter', 1);
+      }
+      if ($show_parent) {
+        $node->set('field_include_parent_org_search', 1);
+      }
+      $node->setSyncing(TRUE);
+      $node->revision_log = "Drush command setting field_org_no_search_filter and field_include_parent_org_search to 1";
+      $node->setRevisionCreationTime(\Drupal::time()->getCurrentTime());
+      $node->save();
+    }
+    catch (\Exception $e) {
+      \Drupal::state()->set('entity_hierarchy_disable_writes', FALSE);
+    }
+    if (!$node->isLatestRevision()) {
+      $storage = \Drupal::entityTypeManager()->getStorage('node');
+      $query = $storage->getQuery()->accessCheck(FALSE);
+      $query->condition('nid', $node->id());
+      $query->latestRevision();
+      $rids = $query->execute();
+      foreach ($rids as $rid) {
```

Comment:
What revision ids are we iterating over here. All the non-published ones? That seems unnecessary to edit all of those. If we are iterating over just one revision, lets be more clear and not use a loop.

Reply:
We are always getting the latest non-publish version, this is the standard code we use in all places, I've refactored it a bit to exclude iteration, since we will always get only 1 result from the query execution